src/librustc/dep_graph/graph.rs

```diff
@@ -948,8 +940,8 @@ struct DepNodeData {
 }
 
 pub(super) struct CurrentDepGraph {
-    data: IndexVec<DepNodeIndex, DepNodeData>,
-    node_to_node_index: FxHashMap<DepNode, DepNodeIndex>,
+    data: Lock<IndexVec<DepNodeIndex, DepNodeData>>,
+    node_to_node_index: Lock<FxHashMap<DepNode, DepNodeIndex>>,
```

Comment:
We should document that the lock for `node_to_node_index` cannot be acquired the lock for `data` is held, or we risk deadlock (since there is code that acquires `node_to_node_index` first).

This is particularly important since there is plenty of random code in this file that seems to grab `current.data.lock` and so forth, so it's not like you can only audit the methods of this particular struct.

(I didn't see any violations, though.)

Also -- given that this change seems to make the logic here that much harder, do we know what kind of benefit we expect? It seems like the primary benefit here would be that operations which only need `node_to_node_index` can run in parallel with those that need `data` -- but I'm not sure if that's important for performance for some specific reason? Maybe we can document that as well (e.g., operation X and Y are able to run in parallel).

Or perhaps what's more important is the `total_read_count` etc below? In that case, let's put the data and node-to-node-index under one lock to eliminate room for error.